### PR TITLE
Use native biometrics when AP fails

### PR DESCRIPTION
**Pull request by jorunfa**
Implemented using https://github.com/fython/BiometricPromptCompat, to get pre Android Pie BiometricPrompt support. Limited traction, but open source & Apache 2 licensed. Seems to be decent quality.

IDBIO-31

### CODE REVIEW

File: app/src/main/java/com/telenordigital/protoman/MainActivity.kt [hunk 1]
```diff
@@ -31,6 +39,41 @@ class MainActivity : AppCompatActivity() {
...
+    private fun promptForBiometrics() {
+        val authenticationCallback = object : BiometricPromptCompat.IAuthenticationCallback {
+            override fun onAuthenticationSucceeded(result: BiometricPromptCompat.IAuthenticationResult) {
+                if (ConnectSdk.getAccessToken() != null) {
+                    goToSignedInActivity()
+                }
+            }
+
+            override fun onAuthenticationHelp(helpCode: Int, helpString: CharSequence?) {
+                // ignore, help text is showed in dialog automatically
```
Comment (metareven) [inline app/src/main/java/com/telenordigital/protoman/MainActivity.kt:51]: Why override this method at all then?

File: app/src/main/java/com/telenordigital/protoman/MainActivity.kt [hunk 2]
```diff
@@ -31,6 +39,41 @@ class MainActivity : AppCompatActivity() {
...
+
+            override fun onAuthenticationFailed() {
+//                Toast.makeText(this@MainActivity, "onAuthenticationFailed", Toast.LENGTH_SHORT).show()
+            }
+        }
+        BiometricPromptCompat.Builder(this)
+                .setTitle(getString(R.string.biometric_auth_title))
+                .setSubtitle(getString(R.string.biometric_auth_subtitle))
+                .setNegativeButton(getString(R.string.biometric_auth_cancel)) { _, _ ->
+//                    Toast.makeText(
```
Comment (metareven) [inline app/src/main/java/com/telenordigital/protoman/MainActivity.kt]: Please remove these comments

**Review by metareven — APPROVED** [review]
OK except for a couple of comments

File: app/src/main/java/com/telenordigital/protoman/MainActivity.kt (hunk 1 above)
Comment (jorunfa) [reply app/src/main/java/com/telenordigital/protoman/MainActivity.kt:51]: It implements an interface - you have to. Or make your class abstract.

File: app/src/main/java/com/telenordigital/protoman/MainActivity.kt (hunk 2 above)
Comment (jorunfa) [reply app/src/main/java/com/telenordigital/protoman/MainActivity.kt]: Will do.

**Pull request merged**